### ISSUE TITLE
Wer macht die Popups beim Erstellen der Figures

### ISSUE THREAD

**Issue by dermondbaer** [g4rgoil/graphing_calculator#60]
Ich kann sie machen, du kannst sie machen ... momentan sind halt keine Methoden in CoordinateSystem, die selbst Popups erzeugen.

**Comment by pmehnert** [comment]
Die sind drin. einfach die create_point, create_distance oder create_line Funktionen aufrufen.

**Comment by pmehnert** [comment]
Du musst halt zuerst syncen.

**Comment by dermondbaer** [comment]
Oke hab den commit nicht bemerkt :)

**Comment by pmehnert** [comment]
Sehen nicht sonderlich schön aus, ABER ICH HATTE NICHT SO VIEL ZEIT FFS.

**Comment by dermondbaer** [comment]
du hast vor allem wieder den Vererbungsbug reingefügt sodass das Koordinatensystem im Hauptfenster liegt ... ich änders commit du mal nix bitte

**Comment by dermondbaer** [comment]
und übernimm das danach bei dir

**Comment by pmehnert** [comment]
nein da ist kein bug.

**Comment by dermondbaer** [comment]
Doch, das hatte ich gefixt. Schau dir einfach mal den Commit an, den ich gleich push ;)

**Comment by pmehnert** [comment]
Juuuunge, das war richtig, das CoordinateSystem ist ein Frame!

**Comment by dermondbaer** [comment]
So, jetzt übernimm das bitte in deinen Branch #63

**Comment by pmehnert** [comment]
Hast du nicht gelesen, was ich grad geschrieben hab? Das war richtig so ffs!

**Comment by pmehnert** [comment]
Reverte das bitte!

**Comment by dermondbaer** [comment]
Nein, war es nicht, sonst hätte ich es nicht fixen müssen. Du hast den falschen Master gewählt, der weitergegebene wurde nicht bis zum Canvas durchgereicht, welches sich deshalb in der Hauptinstanz installiert hat. Das ist nicht gewollt und auch nicht richtig!

**Comment by dermondbaer** [comment]
Es gibt sonst keine Möglichkeit, das in ein eigenes Toplevel-Element einzubinden. Das stimmt schon so. Lass es doch einfach es funktioniert ja ich weiß schon was ich mach.

**Comment by pmehnert** [comment]
Ey, die Klasse CoordinateSystem ist ein Frame, also war es richtig, bei mir läufts.

**Comment by pmehnert** [comment]
Egal, ich hab auf so n scheiß grad kein Bock.

**Comment by dermondbaer** [comment]
Ja, du hast es aber auch immer in das Root-Element eingebettet - hab ich doch schon erklärt. Das Elternobjekt muss logischerweise weitergereicht werden, damit es die Basis für das Canvas sein kann. Es funktioniert doch, also was ist das Problem?

**Comment by pmehnert** [comment]
Wie gesagt mach es halt so.

**Comment by pmehnert** [comment]
Jetzt funktioniert restart nämlich nicht mehr gj.

**Comment by pmehnert** [comment]
Der parameter master soll dabei den nächst 'tieferen Frame markieren. Übergeb einfach self als master.

**Comment by pmehnert** [comment]
Und zwar in der **init** und der restart()

**Comment by dermondbaer** [comment]
Also es ist mir so langsam eigentlich egal, ob du es nachvollziehen kannst, aber: Deine Klasse CoordinateSystem erbt von Frame, was wiederum ein tkinter-Objekt ist. Deswegen brauchst du noch eine Referenz auf das Toplevel-Widget, welches das Elternobjekt für Frame oder Canvas sein soll.

Du kannst natürlich auch noch probieren, ob es ebenfalls funktioniert, wenn du bei der Zeile
`Frame.__init__(self)`
das Frame mit dem Toplevel-Widget als Master initialisierst ... wie das geht, weiß ich nicht, ich hätte die ganze Klasse nicht von Frame abgeleitet und es dann einfacher gehabt, aber machbar ist das bestimmt.

Die Frage ist nur, was es bringt, wenn es auch so funktioniert ...

**Comment by dermondbaer** [comment]
`Frame.__init__(self, self.__master)` funktioniert auch. Ich merge es so in den master, dann sollte auch die restart()-Methode nicht beeinflusst sein.

**Comment by pmehnert** [comment]
Ja viel Spaß noch, ich hab kein Bock mehr 2 Stunden vor Abgabe sowas zu besprechen. Du darfst das jetzt fertig machen und ich esse was. Have fun

**Comment by dermondbaer** [comment]
k

**Comment by pmehnert** [comment]
Schicks mir halt iewann damit ichs abgeben kann. Ansonsten geb ich halt ne alte Version ab.

**Comment by dermondbaer** [comment]
Ich sag dir wann ich fertig bin oder jedenfalls abgabefertig

**Comment by pmehnert** [comment]
K
